### PR TITLE
Adjusting RestAPI dependencies using internal project references

### PR DESCRIPTION
**Pull request by trooper2013**
Fixing crash on iOS while launching. Changed project dependencies to internal project references.

### CODE REVIEW

**Comment by wmathurin** [comment]
We had the dependencies setup that way for ever. Why the crash now?

**Comment by wmathurin** [comment]
Or is that a iOS 12 crash?

**Comment by trooper2013** [comment]
@wmathurin This is the new Swift RestAPIExplorer. The libs were weakly linked. Crashes only on iOS device.

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -20,10 +34,368 @@
...
+			proxyType = 2;
+			remoteGlobalIDString = CE4CE26B1C0E449C009F6029;
+			remoteInfo = SalesforceSDKCore;
+		};
+		B79F06F620EA931900BC7D6F /* PBXContainerItemProxy */ = {
+			isa = PBXContainerItemProxy;
+			containerPortal = B79F06EC20EA931900BC7D6F /* SalesforceSDKCore.xcodeproj */;
+			proxyType = 2;
+			remoteGlobalIDString = 82D4642419C7C8170006BDFE;
+			remoteInfo = SalesforceSDKCoreTestApp;
```
Comment (bhariharan) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer.xcodeproj/project.pbxproj]: Why are we referencing the test targets and the static targets here? A `Swift` app can't use static libs and tests are unnecessary.

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -39,21 +411,115 @@
...
+			name = Products;
+			sourceTree = "<group>";
+		};
+		B79F06DD20EA931200BC7D6F /* Products */ = {
+			isa = PBXGroup;
+			children = (
+				B79F06E520EA931200BC7D6F /* SmartSync.framework */,
+				B79F06E720EA931200BC7D6F /* SmartSyncTestApp.app */,
+				B79F06E920EA931200BC7D6F /* SmartSyncTests.xctest */,
+				B79F06EB20EA931200BC7D6F /* libSmartSync.a */,
```
Comment (bhariharan) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer.xcodeproj/project.pbxproj]: Same question here - why reference the `.a` and test outputs?

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer.xcodeproj/project.pbxproj (hunk 1 above)
Comment (trooper2013) [reply native/SampleApps/RestAPIExplorer/RestAPIExplorer.xcodeproj/project.pbxproj]: @bhariharan Its weird the project settings doesn't show me these dependencies. They shouldn't have gotten added.

**Pull request merged**